service/src/main/java/greencity/service/google/GoogleApiService.java

```diff
@@ -68,4 +75,65 @@ public GeocodingResult getGeocodingResultByCityAndCountryAndLocale(String countr
             throw new GoogleApiException(e.getMessage());
         }
     }
+
+    /**
+     * Retrieves the address details from Google API based on the given geographical
+     * coordinates. This method sends a request to the Google Geocoding API with the
+     * provided latitude and longitude. It processes the response to extract address
+     * components such as city and region. If the API call fails or no results are
+     * found, it returns {@code null}.
+     *
+     * @param coordinates the {@link LatLng} object containing the latitude and
+     *                    longitude.
+     * @return an {@link AddressResponseFromGoogleAPI} containing city, region, and
+     *         district details, or {@code null} if the API call fails or no results
+     *         are returned.
+     */
+    public AddressResponseFromGoogleAPI getResultFromGoogleByCoordinates(LatLng coordinates) {
+        try {
+            GeocodingResult[] results = GeocodingApi.newRequest(context)
+                .latlng(coordinates)
+                .language(UKRAINIAN.getLanguage())
+                .await();
+
+            return Optional.ofNullable(results)
+                .filter(r -> r.length > 0)
+                .map(r -> getAddressResponse(r[0].addressComponents))
```

Comment:
Why do you work only with the first element?
can results contain more than 1 element?

Reply:
Yes, the result contains multiple elements because the Google API returns several results for a given set of coordinates, especially in densely populated regions. However, result[0] is typically the most accurate result.

Reply:
> result[0] is typically the most accurate result.
I see `typically` in your response
What does happen if result[0] is not the most accurate one?